native/native_main.nim

```diff
@@ -0,0 +1,71 @@
+# compile with 
+# nim c --verbosity:0 native_main.nim
+
+# test with
+# time printf '%c\0\0\0{"cmd": "run", "command": "echo $PATH"}' 39 | ./native_main
+
+# compare with
+# time printf '%c\0\0\0{"cmd": "run", "command": "echo $PATH"}' 39 | python ~/.local/share/tridactyl/native_main.py
+
+import json
+import options
+import osproc
+# https://nim-lang.org/docs/json.html
+
+type 
+    MessageResp* = object
+        cmd*, version*, content*, error*, command*: Option[string]
+        code: Option[int]
+
+# let a = MessageResp(cmd: some(""))
+# echo(a)
+
+proc getMessage(): MessageResp =
+
+    # length of the string - not required AFAICT
```

Comment:
We'd need to read only this many bytes if Firefox reuses a single process for multiple messages. But I think we have FF start a new native messenger process for every message.

Reply:
Yeah, we start a new native messenger each time at the moment. We could switch to a daemon based method but I'm hoping this will be quick enough to mean that we don't need to.